src/k8s/cmd/k8s/k8s_bootstrap.go

```diff
@@ -125,16 +143,13 @@ func newBootstrapCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 }
 
 func getConfigFromYaml(filePath string) (apiv1.BootstrapConfig, error) {
-	config := apiv1.BootstrapConfig{}
-	config.SetDefaults()
-
-	yamlContent, err := os.ReadFile(filePath)
+	b, err := os.ReadFile(filePath)
 	if err != nil {
-		return config, fmt.Errorf("failed to read YAML config file: %w", err)
+		return apiv1.BootstrapConfig{}, fmt.Errorf("failed to read file: %w", err)
```

Comment:
this is a different default config from lines 98->114 above.  why are we returning a valid BootstrapConfig at all and not just `nil`?

Reply:
Note that the return signature is `(apiv1.BootstrapConfig, error)`, not `(*apiv1.BootstrapConfig, error)`. In the latter case, we would return `nil` instead, as you said.

Reply:
but good point. the returned `config` would still be empty, but updated the code anyway :+1: